homeassistant/components/home_connect/coordinator.py

```diff
@@ -289,29 +289,13 @@ def _call_all_event_listeners_for_appliance(self, ha_id: str) -> None:
 
     async def _async_update_data(self) -> dict[str, HomeConnectApplianceData]:
         """Fetch data from Home Connect."""
-        try:
-            appliances = await self.client.get_home_appliances()
-        except UnauthorizedError as error:
-            raise ConfigEntryAuthFailed(
-                translation_domain=DOMAIN,
-                translation_key="auth_error",
-                translation_placeholders=get_dict_from_home_connect_error(error),
-            ) from error
-        except HomeConnectError as error:
-            raise UpdateFailed(
-                translation_domain=DOMAIN,
-                translation_key="fetch_api_error",
-                translation_placeholders=get_dict_from_home_connect_error(error),
-            ) from error
```

Comment:
Not updating the appliances on refresh can lead to the following edge cases:
- If an appliance is connected, disconnected, paired or depaired while the event stream is down, we won't get that information
- (This is not very important) If the user updates the appliance name on the API, before it could update the name with `homeassistant.update_entity`, now it only be able to update the name with the integration reload.

Reply:
Right, let's put this back then. :+1:

Reply:
If we omit the second point, I think I've got an idea:
We can refresh it the appliance data this way:
https://github.com/home-assistant/core/blob/40d36275efcd0e231b3edafefd61c656da58f220/homeassistant/components/home_connect/coordinator.py#L270-L273
```diff
            # if there was a non-breaking error, we continue listening
            # but we need to refresh the data to get the possible changes
            # that happened while the event stream was interrupted
+           self._async_setup()
            await self.async_refresh()
```

Reply:
I'm not following. `_async_setup` only makes one request to get all the appliances. Don't we want a complete state update to get the correct state of all entities? Is that handled some other way now?

Reply:
Ok, sounds good.